pkg/api/graph/graph.go

```diff
@@ -122,7 +122,7 @@ var DefaultNamer Namer = namer{}
 func (namer) ResourceName(obj interface{}) string {
 	switch t := obj.(type) {
 	case uniqueNamer:
-		return t.UniqueName()
+		return t.UniqueName().UniqueName()
```

Comment:
i think you should be able to just cast t.UniqueName() to a string?   I guess that's exactly what the second UniqueName() call does.  not sure which is less confusing.